### ISSUE TITLE
localhost actively refused RPC connection

### ISSUE THREAD

**Issue by nerd2ninja** [dgarage/NBXplorer#131]
I have bitcoin core up and running but had to change the directory of the blocks to my hard drive instead of my solid state because of space. Let me know if that would be causing this issue. Otherwise here is the output that I'm getting.

D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer>dotnet run
info: Configuration:  Data Directory: C:\Users\nerd2\AppData\Roaming\NBXplorer\Main
info: Configuration:  Configuration File: C:\Users\nerd2\AppData\Roaming\NBXplorer\Main\settings.config
info: Configuration:  Network: Mainnet
info: Configuration:  Supported chains: BTC
info: Configuration:  Network: Mainnet
info: Configuration:  Supported chains: BTC
info: Configuration:  BTC: Testing RPC connection to http://localhost:8332/
Hosting environment: Production
Content root path: D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer\bin\Debug\netcoreapp2.1\
Now listening on: http://127.0.0.1:24444
Application started. Press Ctrl+C to shut down.
fail: Configuration:  BTC: Error connecting to RPC server No connection could be made because the target machine actively refused it No connection could be made because the target machine actively refused it
fail: Configuration:  BTC: Unhandled in Waiter loop
NBXplorer.Configuration.ConfigException
   at NBXplorer.Configuration.RPCArgs.TestRPCAsync(NBXplorerNetwork networkInfo, RPCClient rpcClient, CancellationToken cancellation) in D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer\Configuration\RPCArgs.cs:line 134
   at NBXplorer.BitcoinDWaiter.StepAsync(CancellationToken token) in D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer\BitcoinDWaiter.cs:line 259
   at NBXplorer.BitcoinDWaiter.StartLoop(CancellationToken token, AutoResetEvent tick) in D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer\BitcoinDWaiter.cs:line 197
Application is shutting down...
^C
D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer>

**Comment by NicolasDorier** [comment]
use `.\run.sh or .\run.ps1`. Read the logs of your bitcoind to know what port it listen for RPC.

**Comment by nerd2ninja** [comment]
I think you're trying to tell me to use the run script instead of typing in powershell or command prompt. The output is the same though. I don't know where the logs for the bitcoin program are. Can you give me a file path starting with the bitcoin folder?

**Comment by NicolasDorier** [comment]
`bitcoind.exe -printtoconsole`

**Comment by nerd2ninja** [comment]
I tried some really creative things and I'm still not getting anything.

D:\Bitcoin>bitcoind.exe -printtoconsole
'bitcoind.exe' is not recognized as an internal or external command,
operable program or batch file.

That prints out no matter what folder I'm in.

D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer>.\run.sh bitcoind.exe -printtoconsole
'.\run.sh' is not recognized as an internal or external command,
operable program or batch file.

I really am trying. Please help

**Comment by nerd2ninja** [comment]
D:\Bitcoin\NBXplorer\NBXplorer\NBXplorer>dotnet run bitcoind.exe -printtoconsole
Specify --help for a list of available options and commands.
fail: Explorer:
      Unrecognized command or argument 'bitcoind.exe'

Even tried this :/

**Comment by NicolasDorier** [comment]
I don't know where your bitcoind install is

**Comment by NicolasDorier** [comment]
`Binding RPC on address ::1 port 8332 failed. `

**Comment by nerd2ninja** [comment]
Alright so, how do I change the RPC port on NBXplorer to 8333 since 8332 is failing.

**Comment by NicolasDorier** [comment]
`bitcoind --help`, `./run.ps1 --help`

**Comment by nerd2ninja** [comment]
Look, I've gotten as far as I have with this vague guidance because I learned the windows command line as a child. I dealt with the pain of trying to learn those new commands from the command line because at the time, I thought it was it fun. Now that I'm older, I don't have time to learn new commands I'll only use once. The internet is full of new projects that like to create new commands people have to learn. 

Do you expect your customer base to be technicians? I am a client side technician so hey got me there and everything but you don't really expect everyone who shows up here to be do you?

In any case, I'm not one of your technicians. Make a standard operating procedure so people who encounter this problem in the future can look back and follow the steps without learning new commands  from the internal help log in command prompt.

Also, its dotnet run --help not ./run.ps1 --help. Either that or you're talking about going back a directory and typing run.ps1 -help which raises another question what is with the unneccesary dashes? I only know to go back a directory by the way, if that is the case, because of my experience and I know that ./ is supposed to go back a directory supposedly. Its never worked for me though I always type cd ..

**Make a standard operating procedure**

**Comment by NicolasDorier** [comment]
Sorry I don't have time to hand hold you, the error message is clear, and the documentation of bitcoind and nbxplorer to setup your own custom rpc port is also clear. Feel free to try another project, but believe me this one is easy to use compared to the rest.

> Also, its dotnet run --help not ./run.ps1 --help

It is `./run.ps1`, not using it may end up in the launch profiles to be applied.

The standard procedure is to run bitcoind without arguments and to run nbxplorer without argument, and it just works. Your machine use the default RPC port for unknown reason, and I don't have time to make a diagnostic of your machine.

**Comment by nerd2ninja** [comment]
@britttttk Thanks for your effort. I've decided to move on though. The reason for all this set up was to get a lightning node up and running. The eclair LN node only took 10 minutes and I don't really have a good reason to continue trying to set up whatever all this is anymore.

Also, I figured I should let you know that you're right kinda. Popped the following into my BTC config file (through notepad and not the command line because that would be absurd):

server=1
txindex=1
rpcuser=USER
rpcpassword=PASSWORD
zmqpubrawblock=tcp://127.0.0.1:29000
zmqpubrawtx=tcp://127.0.0.1:29000
addresstype=p2sh-segwit

NBXplorer now displays the following:
PS D:\bitcoin\nbxplorer\nbxplorer\nbxplorer> dotnet run
info: Configuration:  Data Directory: C:\Users\nerd2\AppData\Roaming\NBXplorer\Main
info: Configuration:  Configuration File: C:\Users\nerd2\AppData\Roaming\NBXplorer\Main\settings.config
info: Configuration:  Network: Mainnet
info: Configuration:  Supported chains: BTC
info: Configuration:  Network: Mainnet
info: Configuration:  Supported chains: BTC
info: Configuration:  BTC: Testing RPC connection to http://localhost:8332/
Hosting environment: Production
Content root path: D:\bitcoin\nbxplorer\nbxplorer\nbxplorer\bin\Debug\netcoreapp2.1\
Now listening on: http://127.0.0.1:24444
Application started. Press Ctrl+C to shut down.
fail: Configuration:  BTC: Error connecting to RPC server The remote server returned an error: (401) Unauthorized.
fail: Configuration:  BTC: Unhandled in Waiter loop
NBXplorer.Configuration.ConfigException
   at NBXplorer.Configuration.RPCArgs.TestRPCAsync(NBXplorerNetwork networkInfo, RPCClient rpcClient, CancellationToken cancellation) in D:\bitcoin\nbxplorer\nbxplorer\nbxplorer\Configuration\RPCArgs.cs:line 134
   at NBXplorer.BitcoinDWaiter.StepAsync(CancellationToken token) in D:\bitcoin\nbxplorer\nbxplorer\nbxplorer\BitcoinDWaiter.cs:line 259
   at NBXplorer.BitcoinDWaiter.StartLoop(CancellationToken token, AutoResetEvent tick) in D:\bitcoin\nbxplorer\nbxplorer\nbxplorer\BitcoinDWaiter.cs:line 197
info: Configuration:  BTC: Testing RPC connection to http://localhost:8332/
fail: Configuration:  BTC: Error connecting to RPC server The remote server returned an error: (401) Unauthorized.
fail: Configuration:  BTC: Unhandled in Waiter loop
NBXplorer.Configuration.ConfigException
   at NBXplorer.Configuration.RPCArgs.TestRPCAsync(NBXplorerNetwork networkInfo, RPCClient rpcClient, CancellationToken cancellation) in D:\bitcoin\nbxplorer\nbxplorer\nbxplorer\Configuration\RPCArgs.cs:line 134
   at NBXplorer.BitcoinDWaiter.StepAsync(CancellationToken token) in D:\bitcoin\nbxplorer\nbxplorer\nbxplorer\BitcoinDWaiter.cs:line 259
   at NBXplorer.BitcoinDWaiter.StartLoop(CancellationToken token, AutoResetEvent tick) in D:\bitcoin\nbxplorer\nbxplorer\nbxplorer\BitcoinDWaiter.cs:line 197
Application is shutting down...

I can assume this merely means that it doesn't know my custom rpc password, but again I've lost my motivation to try to figure out how to configure it.
